components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/NatsInjectHandler.java

```diff
@@ -0,0 +1,151 @@
+                builder = new SOAPBuilder();
+            } else {
+                int index = contentType.indexOf(';');
+                String type = index > 0 ? contentType.substring(0, index) : contentType;
+                builder = BuilderUtil.getBuilderFromSelector(type, axis2MsgCtx);
+                if (builder == null) {
+                    log.warn("No message builder found for type '" + type + "'. Falling back to SOAP.");
+                    builder = new SOAPBuilder();
+                }
+            }
+            // set the message payload to the message context
+            InputStream in = new AutoCloseInputStream(new ByteArrayInputStream((byte[]) object));
+            OMElement documentElement = builder.processDocument(in, contentType, axis2MsgCtx);
+            msgCtx.setEnvelope(TransportUtils.createSOAPEnvelope(documentElement));
+            // Inject the message to the sequence.
+            if (StringUtils.isEmpty(injectingSeq)) {
+                log.error("Sequence name not specified. Sequence : " + injectingSeq);
+                return false;
+            }
+            SequenceMediator seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+            if (seq != null) {
+                printDebugLog("Injecting message to sequence : " + injectingSeq);
+                if (!seq.isInitialized()) {
+                    seq.init(synapseEnvironment);
+                }
+                seq.setErrorHandler(onErrorSeq);
+                return synapseEnvironment.injectInbound(msgCtx, seq, sequential);
+            } else {
+                log.error("Sequence: " + injectingSeq + " not found");
+            }
+        } catch (SynapseException se) {
+            throw se;
+        } catch (Exception e) {
+            throw new SynapseException("Error while processing the NATS Message", e);
+        }
+        printDebugLog("Processed NATS Message: " + new String((byte[])object));
+        return true;
+    }
+
+    /**
+     * Create the initial message context for NATS
+     */
+    private org.apache.synapse.MessageContext createMessageContext() {
+        org.apache.synapse.MessageContext msgCtx = synapseEnvironment.createMessageContext();
+        MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                .getAxis2MessageContext();
+        axis2MsgCtx.setServerSide(true);
+        axis2MsgCtx.setMessageID(UIDGenerator.generateUID());
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        axis2MsgCtx.setProperty(MultitenantConstants.TENANT_DOMAIN, carbonContext.getTenantDomain());
+        msgCtx.setProperty(MessageContext.CLIENT_API_NON_BLOCKING, true);
+        return msgCtx;
+    }
+
+    /**
+     * Check if debug is enabled for logging.
+     *
+     * @param text log text
+     */
+    private void printDebugLog(String text) {
+        if (log.isDebugEnabled()) {
```

Comment:
The reason why we do check for log.isDebugEnabled()) is to avoid the string concatenation that happens while performing log.debug(string). If we introduce this kind of a method, the string concatenation will still happen and there is basically no use of doing log.isDebugEnabled() since this validation anyway happens inside log.debug(string)